File: SHLBlog/shlblog.client/src/app/models/post.ts
import { UserDTO } from "./user";

export class Post
{
  constructor(public PostId: number, public Title: string, public Content: string, public CreatedAt: Date, public UserId: number)
  { }
}

export class PostDTO
{
  constructor(public postId: number, public title: string, public content: string, public createdAt: Date, public userId: number, public userName: UserDTO)
  { }
}

File: SHLBlog/SHLBlog.Server/Models/Post.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SHLBlog.Server.Models
{
    [Table("Posts")]
    public class Post
    {
        public int PostId { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime? CreatedAt { get; set; }
        public int? UserId { get; set; }
        public User? Username { get; set; }
    }
}
